design/06_user_stories.md

```diff
@@ -0,0 +1,31 @@
+# Proposal number 6 : User stories
+
+In order to define the features required to the backup tool, we need user stories that cover use cases for the tool. These should be the point of focus of testing and development. They will represent **what** the code does, in order to create the **how** as a consequence. This proposal attempts to describe them.
+
+One may find this file to be incomplete, as it only represents the feature of a Minimum Viable Product. A backup over the network is not included, as well as an automated way to backup data, or a description of differences between backup. Those may be included in further versions.
+
+## Global configuration
+
+As a user **using the CLI**, I want to be able to:
+
+- `(1)` after plugging in my USB stick, HDD or SSD, create a new entry for it by providing all relevant information (name, type, location).
+- `(2)` edit and delete the properties of any support that has previously been registered.
+
+## Project configuration
+
+`(3)` As a user, I want to be able to initialize or edit a backup strategy for a specific repository by providing all the required configuration of that strategy : physical replication requirements, number of replication requirements, amount of security of said replications. For this repository I can choose to provide with a whitelist and/or a blacklist of security levels for each file/folder that should be included in the backup. The state of the backup should follow the following rules:
+
+- backup statuses are **inherited by default**. Without any instruction for a specific file or folder (through a blacklist/whitelist), the status of a file/folder is the one of its parent repository. If the parent repository is not part of the backup project, the status is determined by the configuration of the project. If the configuration cannot determine if the file/folder is to be backed up, it is `not categorized`
+- backup status are **ordered** according to the following rule: **Not Categorized** is less restrictive that **Ignored** which is less restrictive than **Backed up**. A file/folder cannot have a more restrictive status than its parent repository. For example, a `misc` folder that is `ignored` restricts any file/folder inside it to be `ignored` or `not categorized`.
```

Comment:
I'm not sure to agree to that. 

- In a project **Backed up**, I can ignore files (exclude from backup), but no file can be **Not categorized** (because of inheritance)
- In a project **Not categorized**, anything can happen, we can have projects inside that are not yet categorized, projects that are backed up, and ignored branches of the filesystem 
- In a project **Ignored**, nothing can be backup, so when scanning the filesystem, we can cut any ignored branch. 

Therefore, to me **Uncategorized < Backed up < Ignored**. 

To me, a typical setup would be: 

```
.
├── boot                                # Ignored
├── etc                                 # Ignored
├── home                                # Uncatecorized
│   └──leo-nils                         # Uncatecorized
│      ├── Desktop                      # Uncatecorized
│      ├── Developpement                # Uncatecorized
│      │   ├── ProjectA                 # Tracked
│      │   │   ├── src                  # Tracked
│      │   │   ├── node_modules         # Ignored
│      │   │   ├── package.json         # Tracked
│      │   │   ├── package-lock.json    # Tracked
│      │   │   └── .hibernarc           # (source of tracking)
│      │   ├── ProjectB                 # Uncatecorized
│      │   │   └── wip                  # Uncatecorized
│      │   └── ProjectC                 # Tracked
│      │       ├── src                  # Tracked
│      │       └── .hibernarc           # (source of tracking)
│      ├── Documents                    # Uncatecorized
│      └── Downloads                    # Uncatecorized
├── tmp                                 # Ignored
├── usr                                 # Ignored
└── var                                 # Ignored
```

And in the cli, we might see

```
    Project                                                       Status            Copies    Up to date
---------------------------------------------------------------------------------------------------
    system files (/boot, /etc, /tmp, /usr, /var)                  IGNORED           
🟠  Project A (~/Developpement/ProjectA)                          TRACKED           1/2       1/2
🟢  Project C (~/Developpement/ProjectC)                          TRACKED           3/3       3/3  
⚠️   ProjectB (~/Developpement/ProjectB)                           UNCATEGORIZED     
⚠️   Desktop (~/Desktop)                                           UNCATEGORIZED
⚠️   Documents (~/Documents)                                       UNCATEGORIZED
⚠️   Downloads (~/Downloads)                                       UNCATEGORIZED
```

Reply:
Yes it would mead that indeed. If we dont, we will have no way to  cut the exploration tree and it will take ages to scan for projects... Maybe we need a 2 levels system with lightweight project identification cutting branches as soon as possible, and another system with a gitignore syntax within a project.